kratos/solving_strategies/convergencecriterias/residual_criteria.h

```diff
@@ -404,12 +404,16 @@ class ResidualCriteria
      * @param Rank The rank of the Dof.
      * @return True if the Dof is free, false otherwise.
      */
-    virtual bool IsFreeDof(
+    bool IsFreeDof(
         const DofType& rDof,
         const int Rank
         )
     {
-        return rDof.IsFree();
+        if constexpr (std::is_same<TSparseSpace, UblasSpace<double, CompressedMatrix, Vector>>::value) {
```

Comment:
same comment as above. I would put the if on the outside of the loops where you use this

Reply:
See my previous comment.